### PR TITLE
Use vcpkgs for dependencies on Windows

### PR DESCRIPTION
**Pull request by edenhill**
This addresses the zlib upgrade in #2934 and makes sure we'll have up-to-date dependencies going forward.

It also makes adding additional dependencies easier.

### CODE REVIEW

File: packaging/nuget/packaging.py [hunk 1]
```diff
@@ -327,19 +327,19 @@ def build (self, buildtype):
                                               a.info.get('arch'),
                                               a.info.get('bldtype'))
             if 'toolset' not in a.info:
-                a.info['toolset'] = 'v120'
```
Comment (mhowlett) [inline packaging/nuget/packaging.py:330]: wondering if this affects OOTB compatibility anywhere important, but 2015 was a long time ago...

**Review by mhowlett — APPROVED** [review]

**Review by jliunyu — APPROVED** [review]

File: packaging/nuget/packaging.py (hunk 1 above)
Comment (edenhill) [reply packaging/nuget/packaging.py:330]: Yeah, I'm not completely sure either. But we'll find out

File: src/rdkafka_ssl.c [hunk 2]
```diff
@@ -39,6 +39,8 @@
 #ifdef _WIN32
 #include <wincrypt.h>
 #pragma comment (lib, "crypt32.lib")
+#pragma comment (lib, "libcrypto.lib")
+#pragma comment (lib, "libssl.lib")
```
Comment (mabrarov) [inline src/rdkafka_ssl.c:43]: The names of these libraries (the file names) depend on build configuration. For example, for debug build of OpenSSL with / for MSVC these libraries can be named as libcrypto**d**.lib and libssl**d**.lib.

The CMake project already uses [FindOpenSSL module](https://cmake.org/cmake/help/v3.20/module/FindOpenSSL.html) which finds correct libs itself:

https://github.com/confluentinc/librdkafka/blob/75c43142e1762779f913d60ccb9b9bec931176c7/CMakeLists.txt#L97

so correct libraries should be added into linker options by CMake.

Comment (mabrarov) [reply src/rdkafka_ssl.c:43]: Refer to #5334

**Pull request merged**